File: src/main/scala/resonantinduction/electrical/tesla/ITesla.java
/**
 * 
 */
package resonantinduction.electrical.tesla;

import net.minecraft.tileentity.TileEntity;

/**
 * @author Calclavia
 * 
 */
public interface ITesla
{
	/**
	 * @param transferEnergy - The energy amount in kilojoules.
	 * @param doTransfer - Actually transfer
	 * @return Energy actually transfered.
	 */
	public double teslaTransfer(double transferEnergy, boolean doTransfer);

	public boolean canTeslaTransfer(TileEntity transferTile);

}

File: src/main/scala/edx/electrical/circuit/component/tesla/ITesla.scala
/**
 *
 */
package edx.electrical.circuit.component.tesla

import net.minecraft.tileentity.TileEntity

/**
 * @author Calclavia
 *
 */
abstract trait ITesla
{
  /**
   * @param transferEnergy - The energy amount in kilojoules.
   * @param doTransfer - Actually transfer
   * @return Energy actually transfered.
   */
  def teslaTransfer(transferEnergy: Double, doTransfer: Boolean): Double

  def canTeslaTransfer(transferTile: TileEntity): Boolean
}